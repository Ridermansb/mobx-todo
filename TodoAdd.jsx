import React, { PureComponent } from 'react'

export default class extends PureComponent {

  onSubmitHandle () {

  }

  render() {
    return <form onSubmit={this.onSubmitHandle}>
      <input type="text" />
    </form>
  }
}