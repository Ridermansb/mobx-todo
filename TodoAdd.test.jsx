import React from 'react';
import TodoAdd from './TodoAdd';
import renderer from 'react-test-renderer';

describe('My TodoAdd component', () => {

  test('renders without crashing', () => {
    const wrapper = renderer.create(<TodoAdd />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });


  test('if no todo was written on input on submit should NOT call add on store', () => {

    const wrapper = shallow(<TodoAdd />);

    wrapper.find('form').simulate('submit');

    expect(fetchMock.called('/api/version')).toBeFalsy();

  });

  test('after submit, should call add on store', () => {

    //

  });

  test('after submit, should clear the input');

});
