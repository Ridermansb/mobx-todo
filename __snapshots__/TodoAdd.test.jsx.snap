// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My TodoAdd component renders without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    type="text"
  />
</form>
`;
